diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1624,3 +1706,102 @@ const Value *
     auto MappedPair = mappedValuesBySn[MappedValue->second];
     return ValFromL ? MappedPair.second : MappedPair.first;
 }
+
+/// Try to find a matching instruction that has been moved forward in one of
+/// the basic blocks. If such an instruction is found, the relocation Reloc is
+/// made valid and its begin and end are set to the first and the last
+/// instructions of the block that was skipped during the search.
+/// \param InstL Starting instruction in the first program.
+/// \param InstR Starting instruction in the second program.
+/// \param prog_to_search The program to search a matching instruction in (the
+///                       corresponding iterator will be moved).
+/// \return True if a matching instruction was found, otherwise false.
+bool DifferentialFunctionComparator::findMatchingOpWithOffset(
+        BasicBlock::const_iterator &InstL,
+        BasicBlock::const_iterator &InstR,
+        Program prog_to_search) const {
+    // Choose which instruction will be moved and backup it
+    auto &MovedInst = prog_to_search == Program::First ? InstL : InstR;
+    auto MovedInstBackup = MovedInst;
+
+    auto tryInlineBackup = ModComparator->tryInline;
+
+    // Mark the possible relocation beginning.
+    Reloc.begin = MovedInst;
+
+    auto BBEnd = MovedInst->getParent()->end();
+
+    // Reset the serial counters since InstL and InstR were already compared as
+    // non-equal and start from the following instruction.
+    sn_mapL.erase(&*InstL);
+    sn_mapR.erase(&*InstR);
+    ++MovedInst;
+    while (MovedInst != BBEnd) {
+        auto sn_mapL_backup = sn_mapL;
```

Comment:
`mappedValuesBySn` should get backed up as well since `cmpValues` (i.e., `cmpOperationsWithOperands`) may change it as well.